common/v2/components/TransactionFlow/TxReceipt.tsx

```diff
@@ -76,34 +82,68 @@ export default function TxReceipt({
   const { assets } = useContext(AssetContext);
   const { networks } = useContext(NetworkContext);
   const [txStatus, setTxStatus] = useState(ITxStatus.PENDING);
-  const [displayTxReceipt, setDisplayTxReceipt] = useState(txReceipt as ITxReceipt);
+  const [displayTxReceipt, setDisplayTxReceipt] = useState<ITxReceipt | undefined>(txReceipt);
   const [blockNumber, setBlockNumber] = useState(0);
   const [timestamp, setTimestamp] = useState(0);
 
+  const protectTxContext = useContext(ProtectTxContext);
+  const getProTxValue = ProtectTxUtils.isProtectTxDefined(protectTxContext);
+
   useEffect(() => {
-    const provider = new ProviderHandler(displayTxReceipt.network || txConfig.network);
-    if (blockNumber === 0 && displayTxReceipt.hash) {
+    setDisplayTxReceipt(txReceipt);
+  }, [setDisplayTxReceipt, txReceipt]);
+
+  const [protectTxCounter, setProtectTxCounter] = React.useState(20);
+  useEffect(() => {
```

Comment:
Separation of concerns: I don't believe `TxReceipt` needs to know under which conditions and for how long a user can cancel a transaction. It seems more like the responsibility for `ProtectTxContext`.

Any reason not to move it?

Reply:
No, makes perfect sense 👍